JunQueue.java

```diff
@@ -1,249 +1,65 @@
-                    isFull();
-                    System.out.println(isFull());
-                    break;
-                    
-                case "peek": // DONE
-                    peek(); 
-                    System.out.println(peek());
-                    break;
 
-                case "count": // DONE
-                    System.out.println(trackQueuePosition);
-                    break;
-                    
-                case "display": // DONE
-                    display(); 
-                    break;
-                
-                case "totaldequeue": // DONE
-                    totalDequeue();
-                    break;
-                
-                default:
-                    System.out.println("That is not an option. Please input 'specs' if you need clarification on what you can call.");
-                    System.out.println("Remember each command executes only one time.");
-                    break;
-            }
-        }
-    }
-    
-	//________________________________THIS IS THE START OF THE QUEUE METHODS__________________________________________
-    
-    public void garbageCollection(){
-        if (trackQueuePosition == 0){
-            queue = new String[1]; // prints null
-            queue[0] = null;
-        }
-        else{
-            String[] newqueue = new String[trackQueuePosition * 2];
-            System.arraycopy(queue, 0, newqueue, 0, trackQueuePosition);
-            this.queue = newqueue;
-            }
-        }
-    
     public void enqueue(String value) {
-        //creates list twice as big when necessary
-        if (value.length() != 1) {
-            System.out.println("Not valid input. Must be 1 character only.");
+        if (value.isEmpty()) {
             return;
         }
-        
+
+        //creates list twice as big when necessary
         if (trackQueuePosition == queue.length) {
             String[] newQueue = new String[queue.length * 2];
+            //No extra methods, System.arraycopy is cheating even if the logic is somewhat simple
             System.arraycopy(queue, 0, newQueue, 0, queue.length);
             this.queue = newQueue;
         }
-        
+
         this.queue[trackQueuePosition] = value;
         trackQueuePosition += 1;
     }
-    
-    public String dequeue(){
+
+    public String dequeue() {
         String upNext = this.queue[0];
         if (trackQueuePosition > 0) {
-           for (int i = 0; i < trackQueuePosition - 1; i++){
-               this.queue[i] = this.queue[i + 1];
-           }
-           trackQueuePosition -= 1;
-           this.queue[trackQueuePosition] = null;
-           return upNext;
-        } else {
-            System.out.println("Queue is empty.");
-            return "Queue is empty.";
-        }
-    }
-    
-    public boolean isEmpty(){
-        if (trackQueuePosition == 0){
-            return true;
-        } else {
-            return false;
-        }
-    }
-    
-    public boolean isFull(){

```

Comment:
The internal memory stuff is completely hidden from the person using the method. They don't keep track of the array.